### ISSUE TITLE
how to pull images behind cooperate proxy?

### ISSUE THREAD

**Issue by eagle-dai** [kyma-project/kyma#14147]
**Description**

During kyma installation, in some areas, many images under eu.gcr.io/kyma-project/ cannot be pulled because of some network reasons. However, setting cooperate proxy in both /etc/systemd/system/containerd.service.d/http_proxy.conf and /usr/lib/systemd/system/docker.service seems not fixed the issue.

**Expected result**
During installation, mages under eu.gcr.io/kyma-project/ can be pulled behind cooperate proxy

**Actual result**
During installation, mages under eu.gcr.io/kyma-project/ cannot be pulled behind cooperate proxy

**Steps to reproduce**
run kyma install (in some areas like China mainland)

**Comment by a-thaler** [comment]
Hi @eagle-dai ,
are you using a minikube setup with a kyma1 installation? With minikube we had a lot of troubles, that's why we moved to k3d with kyma2 and do not support minikube anymore, please give that a try.

However, you could try the `--use-hyperkit-vpnkit-sock` on the deprecated `kyma provision minikube` command

**Comment by eagle-dai** [comment]
> Hi @eagle-dai , are you using a minikube setup with a kyma1 installation? With minikube we had a lot of troubles, that's why we moved to k3d with kyma2 and do not support minikube anymore, please give that a try.
> 
> However, you could try the `--use-hyperkit-vpnkit-sock` on the deprecated `kyma provision minikube` command

Hi @a-thaler, thanks for your reply! I followed the guide from https://kyma-project.io/docs/kyma/latest/02-get-started/01-quick-install/. It is kyma2 and k3d.

**Comment by eagle-dai** [comment]
> 

unresolved is a kind of "stable"

**Comment by tobiscr** [comment]
Hi @eagle-dai , I'm taking over your incident. Can you please provide a bit more information about your setup:

* Which OS are you using (name/version, if you are using Mac/BSD/Linux please provide output of `uname -a`)?
* Which version of docker is used (can be figured out by calling `docker version` )?
* What K3D version are you using (please execute `k3d version`)?

Thanks in advance!

**Comment by eagle-dai** [comment]
> Hi @eagle-dai , I'm taking over the issue. Can you please provide a bit more information about your setup:
> 
> * Which OS are you using (name/version, if you are using Mac/BSD/Linux please provide output of `uname -a`)?
> * Which version of docker is used (can be figured out by calling `docker version` )?
> * What K3D version are you using (please execute `k3d version`)?
> * And finally, which Kyma CLI is used (please run `kyma version`)?
> 
> Thanks in advance!


```
$ uname -a
Linux ubuntu 5.13.0-40-generic #45~20.04.1-Ubuntu SMP Mon Apr 4 09:38:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

$ docker version
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.7-0ubuntu5~20.04.2
 Built:             Mon Nov  1 00:34:17 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.7-0ubuntu5~20.04.2
  Built:            Fri Oct 22 00:45:53 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.9-0ubuntu1~20.04.1
  GitCommit:        
 runc:
  Version:          1.0.1-0ubuntu2~20.04.1
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        

$ k3d version
k3d version v5.4.1
k3s version v1.22.7-k3s1 (default)

$ kyma version
{"level":"INFO","time":"2022-08-08T16:17:20.616+0800","message":"Initializing component reconciler 'busola-migrator'"}
Kyma CLI version: 2.1.3
Kyma cluster version: local
```

**Comment by tobiscr** [comment]
Thanks @eagle-dai for the quick reply. I'm going to reproduce your setup and verify how we can get an HTTP proxy properly configured when using Kyma CLI + K3d.

**Comment by tobiscr** [comment]
I tested it on an AWS EC2 machine using Debian (`Linux ...5.10.0-14-cloud-amd64 #1 SMP Debian 5.10.113-1 (2022-04-29) x86_64 GNU/Linux` ) and it turned out that Docker was not considering the proxy properly.

I tried following things (after each change the Docker deamon was restarted):

1. Setting `http_proxy` + `https_proxy` as env-variables: **no success**
2. Setting `export https_proxy=...` in `/etc/default/docker`: **no success**
3. Following [these instructions](https://www.serverlab.ca/tutorials/containers/docker/how-to-set-the-proxy-for-docker-on-ubuntu/) and creating the file `/etc/systemd/system/docker.service.d/proxy.conf` which includes the https-proxy settings: **Docker started to use the proxy**

BUT, make sure your https-proxy is accepting HTTPs connections. E.g. this won't work: `https_proxy=http://my-proxy:1234` and leads to errors like this:

    $> docker search ubuntu
    Error response from daemon: Get https://index.docker.io/v1/search?q=ubuntu&n=25: proxyconnect tcp: tls: first record does not look like a TLS handshake

Your https_proxy has to support the https protocol (e.g. `https_proxy=https://my-proxy:1234`).

**Comment by tobiscr** [comment]
Please let me know if this unblocks you from testing Kyma. Please have also in mind, that we release Kyma 2.5 a few days ago. It's worth to use the latest released version as we are continuously stabilising Kyma and adding new features.

**Comment by eagle-dai** [comment]
> 

Thanks for the info! I will try Kyma 2.5 later

**Comment by eagle-dai** [comment]
> Please let me know if this unblocks you from testing Kyma. Please have also in mind, that we released Kyma 2.5 a few days ago. It's worth to use the latest released version as we are continuously stabilising Kyma and adding new features.

Thanks for the info! I will try Kyma 2.5 later

**Comment by tobiscr** [comment]
hi @eagle-dai : any news from your side in regards to your issue? I would otherwise close this issue at the beginning of next week.
